File: university-service/src/main/java/ua/com/foxminded/courseproject/service/DayScheduleService.java
package ua.com.foxminded.courseproject.service;

import ua.com.foxminded.courseproject.dto.DayScheduleDto;

import java.time.LocalDate;
import java.util.Map;
import java.util.UUID;

public interface DayScheduleService {

    Map<LocalDate, DayScheduleDto> getStudentDaysSchedule(LocalDate startDay, LocalDate endDay, UUID id);

    Map<LocalDate, DayScheduleDto> getTeacherDaysSchedule(LocalDate startDay, LocalDate endDay, UUID id);

    DayScheduleDto getStudentOneDaySchedule(LocalDate date, UUID id);

    DayScheduleDto getTeacherOneDaySchedule(LocalDate date, UUID id);

}

File: university-service/src/main/kotlin/ua/com/foxminded/courseproject/service/DayScheduleService.kt
package ua.com.foxminded.courseproject.service

import java.time.LocalDate
import java.util.UUID
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.dto.DayScheduleDto

interface DayScheduleService {
    fun getStudentDaysSchedule(startDay: LocalDate, endDay: LocalDate, id: UUID): Flux<Pair<LocalDate, DayScheduleDto?>>
    fun getTeacherDaysSchedule(startDay: LocalDate, endDay: LocalDate, id: UUID): Flux<Pair<LocalDate, DayScheduleDto?>>
    fun getStudentOneDaySchedule(date: LocalDate, id: UUID): Mono<Pair<LocalDate,DayScheduleDto?>>
    fun getTeacherOneDaySchedule(date: LocalDate, id: UUID): Mono<Pair<LocalDate,DayScheduleDto?>>
}

